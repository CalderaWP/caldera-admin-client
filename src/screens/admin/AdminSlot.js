import React from 'react';

export default class AdminSlot extends React.Component {
	constructor(props) {
		super(props);
		this.state = {active: false};
		this.handleActive = this.handleActive.bind(this);
		this.handleDeactive = this.handleDeactive.bind(this);
	}

	handleActive() {
		this.setState({active: true});
	}

	handleDeactive() {
		this.setState({active: false});
	}

};
