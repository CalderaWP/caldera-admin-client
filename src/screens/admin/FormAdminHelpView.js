import React from 'react';
import {createSlotFill} from '@wordpress/components'

const {Fill, Slot} = createSlotFill('FormAdminHelpView');

/**
 * Renders the help section with docs/ product links around main admin screen content
 */
export default class FormAdminHelpView extends React.Component {
	static Content = (props) => (
		<Fill name="FormAdminHelpView.Content">
			<div>{props.children}</div>
		</Fill>
	);

	render() {
		return (
			<main>
				<Slot name="FormAdminHelpView.Content">
					{(items) => <div>{items[items.length - 1]}</div>}
				</Slot>
			</main>
		);
	}
}
