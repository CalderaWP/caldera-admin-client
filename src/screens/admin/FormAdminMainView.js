import React from 'react';
import {createSlotFill} from '@wordpress/components'

const {Fill, Slot} = createSlotFill('FormAdminMainView');

/**
 * Renders the main view for the admin screen.
 */
export default class FormAdminMainView extends React.Component {
	static Content = (props) => (
		<Fill name="FormAdminMainView.Content">
			<div>{props.children}</div>
		</Fill>
	);

	render() {
		return (
			<main>
				<Slot name="FormAdminMainView.Content">
					{(items) => <div>{items[items.length - 1]}</div>}
				</Slot>
			</main>
		);
	}
}
