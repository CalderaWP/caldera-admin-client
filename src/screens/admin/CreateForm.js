import React from 'react';
import Toolbar from './Toolbar';
import Viewer from './Viewer';
import AdminSlot from "./AdminSlot";
import {NewForm} from "../../components/NewForm/NewForm";
import {collectionTypes} from "../../types";
import PropTypes from 'prop-types';

export default class CreateForm extends AdminSlot {

	render() {
		const {forms,onCreateForm} = this.props;

		return (
			<div>
				<Toolbar.NavBar
					label="New Form"
					onActive={this.handleActive}
					onDeactive={this.handleDeactive}
					isActive={this.state.active}
				/>
				{this.state.active &&
				<div>

					<Viewer.Content>
						<NewForm onCreate={onCreateForm}/>
					</Viewer.Content>

				</div>
				}
			</div>
		);
	}


}


CreateForm.propTypes = {
	forms: collectionTypes.formsType,
	onCreateForm: PropTypes.func.isRequired
};
CreateForm.defaultProps = {
	forms: {},
};