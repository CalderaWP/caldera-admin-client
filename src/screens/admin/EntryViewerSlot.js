import React from 'react';
import Toolbar from './Toolbar';
import FormAdminView from './FormAdminView';
import AdminSlot from "./AdminSlot";
import types from "../../types";
import PropTypes from 'prop-types';
import {FormEntryViewer} from "../../components/EntryViewer/FormEntryViewer";


export default class EntryViewerSlot extends AdminSlot {

	render() {
		const {form,entries} = this.props;

		return (
			<div>
				<Toolbar.NavBar
					label="Entry Viewer"
					onActive={this.handleActive}
					onDeactive={this.handleDeactive}
					isActive={this.state.active}
				/>
				{this.state.active &&
				<div>

					<FormAdminView.Content>
						<FormEntryViewer
							form={form}
							entries={entries}
						/>
					</FormAdminView.Content>

				</div>
				}
			</div>
		);
	}


}

/**
 * Prop tyeps for CreateFormSlot component
 * @type {{form: shim, entries: shim}}
 */
EntryViewerSlot.propTypes = {
	form: PropTypes.shape(types.formType),
	entries: PropTypes.shape(types.entriesType),
};

EntryViewerSlot.props = {
	form : {
		ID: '',
		name: ''
	},
	field_details: {
		order: {},
		entry_list: {}
	}
}
