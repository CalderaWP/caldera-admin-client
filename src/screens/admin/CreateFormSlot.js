import React from 'react';
import Toolbar from './Toolbar';
import FormAdminMainView from './FormAdminMainView';
import AdminSlot from "./AdminSlot";
import {NewForm} from "../../components/NewForm/NewForm";
import {collectionTypes} from "../../types";
import PropTypes from 'prop-types';

/**
 * The slot for creating forms
 */
export default class CreateFormSlot extends AdminSlot {

	render() {
		const {forms,onCreateForm} = this.props;
		return (
			<div>
				<Toolbar.NavBar
					label="New Form"
					onActive={this.handleActive}
					onDeactive={this.handleDeactive}
					isActive={this.state.active}
				/>
				{this.state.active &&
				<div>

					<FormAdminMainView.Content>
						<NewForm
							onCreate={onCreateForm}
							forms={forms}
						/>
					</FormAdminMainView.Content>

				</div>
				}
			</div>
		);
	}


}


CreateFormSlot.propTypes = {
	forms: collectionTypes.formsType,
	onCreateForm: PropTypes.func.isRequired
};
CreateFormSlot.defaultProps = {
	forms: {},
};