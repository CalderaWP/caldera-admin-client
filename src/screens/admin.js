import AdminSlot from './admin/AdminSlot';
import FormSlot from './admin/FormsSlot';
import EntryViewerSlot from './admin/EntryViewerSlot';
import SettingsSlot from './admin/SettingsSlot';

export default {
	AdminSlot,
	FormSlot,
	EntryViewerSlot,
	SettingsSlot
};