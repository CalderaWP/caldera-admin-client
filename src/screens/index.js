import admin from './admin';

/**
 * Export screens
 */
export default {
	admin
};
