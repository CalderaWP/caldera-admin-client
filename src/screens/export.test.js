import screens from './index';
import admin from './admin';
import {snapshotObjectKeysAndTypes} from '../testUtil/snapshotObjectKeysAndTypes';

describe( 'screens export', () => {
	it( 'index has the right types and keys', () => {
		snapshotObjectKeysAndTypes(screens);
	});
});

describe( 'admin screen export', () => {
	it( 'admin has the right types and keys', () => {
		snapshotObjectKeysAndTypes(admin);
	});
});