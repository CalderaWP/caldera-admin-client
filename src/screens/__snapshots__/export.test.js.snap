// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin screen export admin has the right types and keys 1`] = `"{\\"AdminSlot\\":\\"function\\",\\"FormSlot\\":\\"function\\",\\"EntryViewerSlot\\":\\"function\\",\\"SettingsSlot\\":\\"function\\"}"`;

exports[`screens export index has the right types and keys 1`] = `"{\\"admin\\":\\"object\\"}"`;
